src/response_decoder.ts

```diff
@@ -272,13 +348,40 @@ export class BufferDataReader implements DataReader {
     return value
   }
 
+  readUTF8String(): string {
+    const type = this.readUInt8()
+    switch (type) {
+      case FormatCode.Str8:
+      case FormatCode.Sym8:
+        const sizeStr8 = this.readUInt8()
+        const valueStr8 = this.data.toString("utf8", this.offset, this.offset + sizeStr8)
+        this.offset += sizeStr8
+        return valueStr8
+      case FormatCode.Str32:
+        const sizeStr32 = this.readUInt32()
+        const valueStr32 = this.data.toString("utf8", this.offset, this.offset + sizeStr32)
+        this.offset += sizeStr32
+        return valueStr32
+      default:
+        throw new Error("ReadUTFString ERROR, unknown string type")
+    }
+  }
```

Comment:
I'm not 100% sure to move this function nin this class because we are creating a coupling between BufferDataReader and show the string are stored in AMQP 1.0 format